### DISCUSSION TITLE
Is there a way to let MSBuild() find the right toolversion with VS 2022 installed using legacy Cake nuget?

### DISCUSSION THREAD

**Discussion by janniksam** [cake-build/cake#3785] [Q&A] [answered]
I am using using the legacy Cake-Nuget and restore it on build, then run Cake.exe
I cant do that anymore because Cake.exe is not content of Cake.Tool.

Since I dont want to rewrite my whole build script I would like to know, if there is any way to get MSBuild() with 2022 working. I am on Cake 1.3.0 right now.

Thanks!

**Comment by augustoproiete** [comment, marked as answer]
@janniksam Upgrading from Cake v1.3 to Cake v2.0 (and use .NET Tool) is probably easier than you think it is, and likely not a rewrite. Check out the steps laid out by @nils-a here: https://github.com/cake-build/cake/discussions/3744#discussion-3733915

But to answer your question, an easy way to use MSBuild / VS2022 in older versions of Cake is by using `vswhere` to locate where MSBuild is installed. There's an example by @DrewRhoades here: https://github.com/cake-build/cake/issues/3671#issuecomment-990390959

**Reply by janniksam** [reply]
Thank you for the quick answer.

The problem that I have is that we use the webploy addin and it doesnt seem to work right now with the newest version. Also it hasnt updated for months now.

I will try to use the vswhere approach and get back to you :)
